pkg/ebpf/c/bpf_telemetry.h

```diff
@@ -77,19 +70,25 @@ static void *(*bpf_telemetry_update_patch)(unsigned long, ...) = (void *)PATCH_T
                     errno_slot &= (T_MAX_ERRNO - 1);                                            \
                 }                                                                               \
                 errno_slot &= (T_MAX_ERRNO - 1);                                                \
-                if (helper_indx >= 0) {                                                         \
-                    long *target = &entry->err_count[(helper_indx * T_MAX_ERRNO) + errno_slot]; \
-                    unsigned long add = 1;                                                      \
-                    /* Patched instruction for 4.14+: __sync_fetch_and_add(target, 1);
-                     * This patch point is placed here because the above instruction
-                     * fails on the 4.4 verifier. On 4.4 this instruction is replaced
-                     * with a nop: r1 = r1 */                                                   \
-                    bpf_telemetry_update_patch((unsigned long)target, add);                     \
-                }                                                                               \
+                __sync_fetch_and_add(&tb->helper_err_telemetry[telemetry_program_id].err_count[(helper_indx * T_MAX_ERRNO) + errno_slot], 1); \
             }                                                                                   \
         }                                                                                       \
         errno_ret;                                                                              \
     })
+// If instrumentation is not enabled do not waste instructions
+#else
+#define map_update_with_telemetry(fn, map, args...) \
+    ({                                              \
+        long errno_ret = fn(&map, args);            \
+        errno_ret;                                  \
+    })
```

Comment:
Can these macros be simplified?
```suggestion
#define map_update_with_telemetry(fn, map, args...) fn(&map, args)
```